extensions/blocks/social-previews/modal.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * Social Previews modal component.
+ *
+ * Shows individual previews in modal window.
+ */
+
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Button, Modal, TabPanel } from '@wordpress/components';
+import { withSelect } from '@wordpress/data';
+import { applyFilters } from '@wordpress/hooks';
+import { has } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { AVAILABLE_SERVICES } from './constants';
+
+const SocialPreviewsModal = function SocialPreviewsModal( props ) {
+	const {
+		onClose,
+		showUpgradeNudge,
+		media,
+		currentPostId,
+		title,
+		description,
+		url,
+		author,
+	} = props;
```

Comment:
Since `props` doesn't seem to be used, you could desestructurante the object straight in the function arguments

```suggestion
const SocialPreviewsModal = function SocialPreviewsModal( {
	onClose,
	showUpgradeNudge,
	media,
	currentPostId,
	title,
	description,
	url,
	author,
) {
```